### ISSUE TITLE
500 Server Error: Internal Server Error for url: https://m.facebook.com/settings?locale=en_US

### ISSUE THREAD

**Issue by cwizard** [kevinzg/facebook-scraper#763]
Just started receiving this error today, we only use this tool to scrape one page so I am not sure where to begin troubleshooting because the error seems to reference something that is part of the module, and not anything I've defined

500 Server Error: Internal Server Error for url: https://m.facebook.com/settings?locale=en_US

**Comment by neon-ninja** [comment]
What code did you use that caused this error? Did you get any other errors along with this?

**Comment by cwizard** [comment]
It's a straightforward call to the plugin, but did you notice that page is just down?  Are you saying that it works for you currently?

**Comment by cwizard** [comment]
Sorry, something went wrong.
We are working on it and we'll get it fixed as soon as we can. -- Maybe something broken on FB side

**Comment by neon-ninja** [comment]
It's working fine for me

**Comment by cwizard** [comment]
Oh wow, that's great lol.  I'm troubleshooting this side, it seems as if I can access that mobile settings page from one machine but not another.  The python itself runs from a contabo linux host.  I'm not asking for that settings page, I assumed it was calling that internally for some reason.  Is that necessary?

**Comment by cwizard** [comment]
Okay I took the cookies from the browser/session that works and that has resolved my original issue.  Now to figure out the reason why that is happening and resolve the cause =D

**Comment by neon-ninja** [comment]
Yes, the scraper hits that settings page to check that the cookies are working ok when you set them - 

https://github.com/kevinzg/facebook-scraper/blob/142433885a265a4bfa49d8fa19143b473747fe1f/facebook_scraper/facebook_scraper.py#L951

https://github.com/kevinzg/facebook-scraper/blob/142433885a265a4bfa49d8fa19143b473747fe1f/facebook_scraper/__init__.py#L50

I would suspect that most regular users don't connect to Facebook from contabo linux hosts, so Facebook are blocking that IP

**Comment by cwizard** [comment]
Dumb question can I incorporate a proxy pool

**Comment by neon-ninja** [comment]
Yes, you can use the `set_proxy` function. However, if your proxy also runs in a cloud datacentre, you'd have the same problem

**Comment by cwizard** [comment]
Well, this problem persists on every host I try to use now and I'm not sure why.,  I check the page once every like..... 4 to 9 minutes on a timer.

**Comment by neon-ninja** [comment]
You're probably triggering Facebook's bot detection. Scrape slower or less

**Comment by BurnsDillion** [comment]
@cwizard I was having this same error. Not sure if this will help you, but be sure you include the mobile post link. I originally used a post URL for regular Facebook, however, I would get that error. Once I found the URL for the m.facebook version of the post, it was able to scrape. Hope this helps you!

**Comment by viseshrp** [comment]
I noticed this too, and you were right. It was the bot detection. I changed scraping frequency from every 45 minutes to 90 and then also generated a new set of cookies. That fixed it for me.
